src/main/java/ca/bc/gov/backendstartapi/endpoint/VegetationCodeEndpoint.java

```diff
@@ -34,7 +33,11 @@
 @Tag(name = "vegetationCode", description = "Codes describing various vegetation species")
 public class VegetationCodeEndpoint {
 
-  @Autowired private VegetationCodeRepository vegetationCodeRepository;
+  private final VegetationCodeRepository vegetationCodeRepository;
+
+  VegetationCodeEndpoint(VegetationCodeRepository vegetationCodeRepository) {
+    this.vegetationCodeRepository = vegetationCodeRepository;
+  }
```

Comment:
You need to annotate the constructor with `@Autowired` here.

Reply:
You don't need to if the bean has only one constructor; see [here](https://docs.spring.io/spring-boot/docs/3.0.x/reference/html/using.html#using.spring-beans-and-dependency-injection).

Reply:
Yes, you're right. As a note, personally I like to annotate the class with lombok's `@NoArgsConstructor` and annotate the right one with `@Autowired`. But that's optional. I think we're good to go here.